### PR TITLE
Switch Webhook to ed25519 and expiration of the certificate to seven days

### PR DESCRIPTION
**Pull request by Harwayne**
# Changes

- :broom: Webhook uses `ed25519` rather than RSA 2048.
- :broom: Change the expiration of the certificate to seven days and the recreation time to one day before expiration.

/kind enhancement

**Release Note**

```release-note
NONE
```

**Docs**

```docs

```

### CODE REVIEW

File: webhook/certificates/certificates_test.go [hunk 1]
```diff
@@ -127,7 +127,7 @@ func TestReconcile(t *testing.T) {
 		Name: "certificate expiring soon",
 		Key:  key,
 		// 6 days falls inside of the grace period of 7 days so the secret will be updated
```
Comment (dprotaso) [inline webhook/certificates/certificates_test.go:129]: update comment

**Review by dprotaso — CHANGES_REQUESTED** [review]
cc @markusthoemmes @julz are security focused folks

File: webhook/certificates/certificates_test.go (hunk 1 above)
Comment (Harwayne) [reply webhook/certificates/certificates_test.go]: Done.

**Review by dprotaso — APPROVED** [review]

**Pull request merged**